src/api_server/src/lib.rs

```diff
@@ -486,31 +640,33 @@ mod tests {
         let (api_request_sender, _from_api) = channel();
         let (to_api, vmm_response_receiver) = channel();
         let seccomp_filters = get_filters(SeccompConfig::Advanced).unwrap();
-        let (socket_ready_sender, socket_ready_receiver) = channel();
 
         thread::Builder::new()
             .name("fc_api_test".to_owned())
             .spawn(move || {
-                ApiServer::new(api_request_sender, vmm_response_receiver, to_vmm_fd)
-                    .bind_and_run(
-                        PathBuf::from(api_thread_path_to_socket),
-                        ProcessTimeReporter::new(Some(1), Some(1), Some(1)),
-                        seccomp_filters.get("api").unwrap(),
-                        vmm::HTTP_MAX_PAYLOAD_SIZE,
-                        socket_ready_sender,
-                    )
-                    .unwrap();
+                ApiServer::new(
+                    api_request_sender,
+                    vmm_response_receiver,
+                    to_vmm_fd,
+                    Default::default(),
+                )
+                .bind_and_run(
+                    PathBuf::from(api_thread_path_to_socket),
+                    ProcessTimeReporter::new(Some(1), Some(1), Some(1)),
+                    seccomp_filters.get("api").unwrap(),
+                    vmm::HTTP_MAX_PAYLOAD_SIZE,
+                )
+                .unwrap();
             })
             .unwrap();
 
-        // Wait for the server to set itself up.
-        socket_ready_receiver.recv().unwrap();
         to_api
             .send(Box::new(Ok(VmmData::InstanceInformation(
                 InstanceInfo::default(),
             ))))
             .unwrap();
-        let mut sock = UnixStream::connect(PathBuf::from(path_to_socket)).unwrap();
+        let mut sock =
+            try_connect(PathBuf::from(path_to_socket), 5, Duration::from_millis(10)).unwrap();
```

Comment:
Interesting tweak, was there previously a risk of this locking up the test until the test framework forcefully timed out?

Reply:
Not locking up, `UnixStream::connect` would sometimes just return error (test fail) because the socket doesn't exist at provided path.

This would happen when the test thread is faster than the spawned API thread.

---

It didn't happen before because `socket_ready_receiver` channel was used as a barrier to wait for the API thread to set up its socket. This "barrier" was used outside of tests as well when the main thread would send "/shutdown-internal" to the API thread.

Now that we removed the public facing "/shutdown-internal" HTTP command, and we shut down using actual _internal mechanisms_, I don't see the point in keeping extra `socket_ready_receiver` channel just for tests.

In tests we can just retry the connection a few times and it will be available if all is well, it will correctly fail if smth is wrong.

Reply:
Thanks for explaining your thinking there. Certainly a leaner approach in my opinion as well.